bvm/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/CPU.java

```diff
@@ -3444,84 +3444,101 @@ private static void convertMapToStruct(WorkerExecutionContext ctx, int[] operand
+                    e.getMessage();
+            handleTypeConversionError(ctx, sf, j, errorMsg);
+        }
+    }
+
+    private static BMap<String, BValue> convertMapToStruct(WorkerExecutionContext ctx,
+                                                           BMap<String, BValue> bMap, BStructureType structType) {
         BMap<String, BValue> bStruct = new BMap<>(structType);
-        StructureTypeInfo structInfo = ctx.callableUnitInfo
-                .getPackageInfo().getStructInfo(structType.getName());
+        StructureTypeInfo structInfo = ctx.callableUnitInfo.getPackageInfo().getStructInfo(structType.getName());
 
         for (StructFieldInfo fieldInfo : structInfo.getFieldInfoEntries()) {
             String key = fieldInfo.getName();
             BType fieldType = fieldInfo.getFieldType();
             BValue mapVal = null;
-            try {
-                boolean containsField = bMap.hasKey(key);
-                DefaultValueAttributeInfo defaultValAttrInfo = null;
-                if (containsField) {
-                    mapVal = bMap.get(key);
-                    if (mapVal == null && BTypes.isValueType(fieldType)) {
-                        throw BLangExceptionHelper.getRuntimeException(
-                                RuntimeErrors.INCOMPATIBLE_FIELD_TYPE_FOR_CASTING, key, fieldType, null);
+
+            boolean containsField = bMap.hasKey(key);
+            DefaultValueAttributeInfo defaultValAttrInfo = null;
+            if (containsField) {
+                mapVal = bMap.get(key);
+                if (mapVal == null && BTypes.isValueType(fieldType)) {
+                    throw BLangExceptionHelper.getRuntimeException(
+                            RuntimeErrors.INCOMPATIBLE_FIELD_TYPE_FOR_CASTING, key, fieldType, null);
+                }
+
+                if (mapVal != null) {
+                    if (mapVal.getType().getTag() == TypeTags.MAP_TAG) {
+                        bStruct.put(key, convertMap(ctx, (BMap<String, BValue>) mapVal, fieldType, key));
+                        continue;
                     }
 
-                    if (mapVal != null && !checkCast(mapVal, fieldType, new ArrayList<TypePair>())) {
+                    if (!checkCast(mapVal, fieldType, new ArrayList<TypePair>())) {
                         throw BLangExceptionHelper.getRuntimeException(
                                 RuntimeErrors.INCOMPATIBLE_FIELD_TYPE_FOR_CASTING, key, fieldType, mapVal.getType());
                     }
-                } else {
-                    defaultValAttrInfo = (DefaultValueAttributeInfo) getAttributeInfo(fieldInfo,
-                            AttributeInfo.Kind.DEFAULT_VALUE_ATTRIBUTE);
                 }
-
-                switch (fieldType.getTag()) {
-                    case TypeTags.INT_TAG:
-                        if (containsField) {
-                            bStruct.put(key, mapVal);
-                        } else if (defaultValAttrInfo != null) {
+            } else {
```

Comment:
We can remove the 'else' block, and take it out. It will reduce the indentations..